src/bptl/credentials/forms.py

```diff
@@ -0,0 +1,95 @@
+from typing import Any, Dict, List
+
+from django import forms
+from django.db.models import BLANK_CHOICE_DASH
+from django.utils.translation import gettext_lazy as _
+
+from zgw_consumers.concurrent import parallel
+from zgw_consumers.constants import APITypes
+from zgw_consumers.models import Service
+
+from bptl.work_units.zgw.utils import get_paginated_results
+
+from .models import App
+
+
+def _get_applications(service: Service) -> List[Dict[str, Any]]:
+    client = service.build_client()
+    results = get_paginated_results(client, "applicatie")
+    return results
+
+
+class AppForm(forms.ModelForm):
+    autorisaties_application = forms.ChoiceField(
+        label=_("Autorisaties API app"),
+        required=False,
+        help_text=_(
+            "Application in Autorisaties API. If selected, label and App ID "
+            "can be left blank."
+        ),
+    )
+
+    class Meta:
+        model = App
+        fields = (
+            "autorisaties_application",
+            "label",
+            "app_id",
+        )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._set_auth_api_choices()
+
+    def _set_auth_api_choices(self):
+        autorisatie_apis = Service.objects.filter(api_type=APITypes.ac)
+
+        with parallel() as executor:
+            applications = executor.map(_get_applications, autorisatie_apis)
+
+        all_apps = sum(applications, [])
+        all_apps = sorted(all_apps, key=lambda app: app["label"])
+
+        # if there's no data, no point in showing the field at all
+        if not all_apps:
+            del self.fields["autorisaties_application"]
+
+        self.fields["autorisaties_application"].choices = BLANK_CHOICE_DASH + [
+            (application["url"], application["label"]) for application in all_apps
+        ]
+        self.fields["label"].required = False
+        self.fields["app_id"].required = False
+
+        if self.instance.pk:
```

Comment:
if there is no field, can there be an initial of that field?

Reply:
the DB value is always loaded as initial for `label` and `app_id`. We "conveniently" use the auth API identifier as the app_id, but technically app_id can be any string.

There is no DB field tracking which auth API application this points to, so there can only be an initial value if we have an existing DB record.

Reply:
Handled by `return`-ing after field deletion - this code will only kick in when there are applications retrieved from the Autorisaties API and the field is therefore present.